## FIFA Football cup

In [1]:
import pandas as pd


### Dataset name : WorldCupMatches

In [5]:
df1 = pd.read_csv('D:\Final submission\EDA\WorldCup-EDA\Dataset\WorldCupMatches.csv')

The FIFA World Cup is a global football competition contested by the various football-playing nations of the world. It is contested every four years and is the most prestigious and important trophy in the sport of football.

The World Cups dataset show all information about all the World Cups in the history, while the World Cup Matches dataset shows all the results from the matches contested as part of the cups.

1. year : The year in which the match was played.
2. Datetime : The date at which match was played along with 24 hrs date format.
3. Stage : The stage at which the match was played.
4. Stadium : Satdium name where the match was held.
5. city : The city name where the match was played.
6. Home team name : Home team country name.
7. Home team gaol : Total goals scored by the home team by the end of the match.
8. Away Team Goals : Total goals scored by the away team by the end of the match.
9. Away Team Name : Away team country name.
10. Win conditions : Special win condition (if any)
11. Attendance : Total crowd present at the satdium
12. Half-time Home Goals : Goals scored by the home team until half time
13. Half-time Away Goals : Goals scored by the away team until half time
14. Referee : Name of the first refree
15. Assistant 1 : Name of the first assistant referee (linesman)
16. Assistant 2 : Name of the second assistant referee (linesman)
17. RoundID : Unique ID of the Round
18. MatchID : Unique ID of the match
19. Home Team Initials : Home team country's three letter initials
20. Away Team Initials : Away team country's three letter initials

### Dataset 2 : WorldCupPlayers

In [6]:
df2 = pd.read_csv('D:\Final submission\EDA\WorldCup-EDA\Dataset\WorldCupPlayers.csv')

##### data information

1. RoundID : Unique ID of the round
2. MatchID : Unique ID of the match
3. Team Initials : Player's team initials
4. Coach Name : Name and country of the team coach
5. Line-up : S=Line-up, N=Substitute
6. Shirt Number : Shirt number if available 
7. Player Name : Name of the player
8. Position: C=Captain, GK=Goalkeeper 
9. Event : G=Goal, OG=Own Goal, Y=Yellow Card, R=Red Card, SY = Red Card by second yellow, P=Penalty, MP=Missed Penalty, I = Substitution In, O=Substitute Out 


### Dataset 3: WorldCups

In [7]:
df3 = pd.read_csv('D:\Final submission\EDA\WorldCup-EDA\Dataset\WorldCups.csv')

#### Dataset information

1. Year : Year of the worldcup
2. Country : Country of the worldcup
3. Winner : Team who won the worldcup
4. Runners-Up : Team who was the second place
5. Third : Team who was the third place
6. Fourth : Team who was the fourth place
7. GoalsScored : Total goals scored in the worldcup
8. QualifiedTeams : Total participating teams
9. MatchesPlayed : Total matches played in the cup
10. Attendance : Total attendance of the worldcup


Here I have 3 different DataFrame and I need to work on all the data frame . The step that are applied on all the above are as follows:
1. Feature eng.
2. Feature selection

Creat a new data frame based on the information

3. Checking the data information
4. Handling Missing values 
5. 5 number summary 



In [8]:
# checking the shape of all the 3 different data frame

df1.shape, df2.shape, df3.shape 

((4572, 20), (37784, 9), (20, 10))

In [9]:
# Checking all the columns of 3 different data frame

df1.columns, df2.columns, df3.columns

(Index(['Year', 'Datetime', 'Stage', 'Stadium', 'City', 'Home Team Name',
        'Home Team Goals', 'Away Team Goals', 'Away Team Name',
        'Win conditions', 'Attendance', 'Half-time Home Goals',
        'Half-time Away Goals', 'Referee', 'Assistant 1', 'Assistant 2',
        'RoundID', 'MatchID', 'Home Team Initials', 'Away Team Initials'],
       dtype='object'),
 Index(['RoundID', 'MatchID', 'Team Initials', 'Coach Name', 'Line-up',
        'Shirt Number', 'Player Name', 'Position', 'Event'],
       dtype='object'),
 Index(['Year', 'Country', 'Winner', 'Runners-Up', 'Third', 'Fourth',
        'GoalsScored', 'QualifiedTeams', 'MatchesPlayed', 'Attendance'],
       dtype='object'))

From the above we can see that the names of the columns so we can check them independently.

### Dataset 1 

In [10]:
df1.head(5)

Year              Datetime    Stage         Stadium         City  \
0  1930.0  13 Jul 1930 - 15:00   Group 1         Pocitos  Montevideo    
1  1930.0  13 Jul 1930 - 15:00   Group 4  Parque Central  Montevideo    
2  1930.0  14 Jul 1930 - 12:45   Group 2  Parque Central  Montevideo    
3  1930.0  14 Jul 1930 - 14:50   Group 3         Pocitos  Montevideo    
4  1930.0  15 Jul 1930 - 16:00   Group 1  Parque Central  Montevideo    

  Home Team Name  Home Team Goals  Away Team Goals Away Team Name  \
0         France              4.0              1.0         Mexico   
1            USA              3.0              0.0        Belgium   
2     Yugoslavia              2.0              1.0         Brazil   
3        Romania              3.0              1.0           Peru   
4      Argentina              1.0              0.0         France   

  Win conditions  Attendance  Half-time Home Goals  Half-time Away Goals  \
0                     4444.0                   3.0                   0.0   
1                    18346.0                   2.0                   0.0   
2                    24059.0                   2.0                   0.0   
3                     2549.0                   1.0                   0.0   
4                    23409.0                   0.0                   0.0   

                  Referee               Assistant 1  \
0  LOMBARDI Domingo (URU)     CRISTOPHE Henry (BEL)   
1       MACIAS Jose (ARG)  MATEUCCI Francisco (URU)   
2     TEJADA Anibal (URU)   VALLARINO Ricardo (URU)   
3   WARNKEN Alberto (CHI)       LANGENUS Jean (BEL)   
4     REGO Gilberto (BRA)      SAUCEDO Ulises (BOL)   

                  Assistant 2  RoundID  MatchID Home Team Initials  \
0         REGO Gilberto (BRA)    201.0   1096.0                FRA   
1       WARNKEN Alberto (CHI)    201.0   1090.0                USA   
2         BALWAY Thomas (FRA)    201.0   1093.0                YUG   
3    MATEUCCI Francisco (URU)    201.0   1098.0                ROU   
4  RADULESCU Constantin (ROU)    201.0   1085.0                ARG   

  Away Team Initials  
0                MEX  
1                BEL  
2                BRA  
3                PER  
4                FRA

In [11]:
df1.shape

(4572, 20)

From the above I can see that there are 4572 rows and 20 columns.

In [12]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4572 entries, 0 to 4571
Data columns (total 20 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Year                  852 non-null    float64
 1   Datetime              852 non-null    object 
 2   Stage                 852 non-null    object 
 3   Stadium               852 non-null    object 
 4   City                  852 non-null    object 
 5   Home Team Name        852 non-null    object 
 6   Home Team Goals       852 non-null    float64
 7   Away Team Goals       852 non-null    float64
 8   Away Team Name        852 non-null    object 
 9   Win conditions        852 non-null    object 
 10  Attendance            850 non-null    float64
 11  Half-time Home Goals  852 non-null    float64
 12  Half-time Away Goals  852 non-null    float64
 13  Referee               852 non-null    object 
 14  Assistant 1           852 non-null    object 
 15  Assistant 2          

From the above i can see that there are names of the column and there not null data and its data type

In [13]:
df1.describe()

Year  Home Team Goals  Away Team Goals     Attendance  \
count   852.000000       852.000000       852.000000     850.000000   
mean   1985.089202         1.811033         1.022300   45164.800000   
std      22.448825         1.610255         1.087573   23485.249247   
min    1930.000000         0.000000         0.000000    2000.000000   
25%    1970.000000         1.000000         0.000000   30000.000000   
50%    1990.000000         2.000000         1.000000   41579.500000   
75%    2002.000000         3.000000         2.000000   61374.500000   
max    2014.000000        10.000000         7.000000  173850.000000   

       Half-time Home Goals  Half-time Away Goals       RoundID       MatchID  
count            852.000000            852.000000  8.520000e+02  8.520000e+02  
mean               0.708920              0.428404  1.066177e+07  6.134687e+07  
std                0.937414              0.691252  2.729613e+07  1.110572e+08  
min                0.000000              0.000000  2.010000e+02  2.500000e+01  
25%                0.000000              0.000000  2.620000e+02  1.188750e+03  
50%                0.000000              0.000000  3.370000e+02  2.191000e+03  
75%                1.000000              1.000000  2.497220e+05  4.395006e+07  
max                6.000000              5.000000  9.741060e+07  3.001865e+08

These describe function will give the infromation about all the statistics 

In [14]:
df1.isnull().sum()

Year                    3720
Datetime                3720
Stage                   3720
Stadium                 3720
City                    3720
Home Team Name          3720
Home Team Goals         3720
Away Team Goals         3720
Away Team Name          3720
Win conditions          3720
Attendance              3722
Half-time Home Goals    3720
Half-time Away Goals    3720
Referee                 3720
Assistant 1             3720
Assistant 2             3720
RoundID                 3720
MatchID                 3720
Home Team Initials      3720
Away Team Initials      3720
dtype: int64

In [16]:
# check the missing percentage 
percentage_missing_value = df1.isnull().sum()/len(df1) * 100
percentage_missing_value.round(2)

Year                    81.36
Datetime                81.36
Stage                   81.36
Stadium                 81.36
City                    81.36
Home Team Name          81.36
Home Team Goals         81.36
Away Team Goals         81.36
Away Team Name          81.36
Win conditions          81.36
Attendance              81.41
Half-time Home Goals    81.36
Half-time Away Goals    81.36
Referee                 81.36
Assistant 1             81.36
Assistant 2             81.36
RoundID                 81.36
MatchID                 81.36
Home Team Initials      81.36
Away Team Initials      81.36
dtype: float64

In [17]:
df1.dropna(inplace=True)

In [19]:
df1.shape

(850, 20)

In [20]:
df1.head()

Year              Datetime    Stage         Stadium         City  \
0  1930.0  13 Jul 1930 - 15:00   Group 1         Pocitos  Montevideo    
1  1930.0  13 Jul 1930 - 15:00   Group 4  Parque Central  Montevideo    
2  1930.0  14 Jul 1930 - 12:45   Group 2  Parque Central  Montevideo    
3  1930.0  14 Jul 1930 - 14:50   Group 3         Pocitos  Montevideo    
4  1930.0  15 Jul 1930 - 16:00   Group 1  Parque Central  Montevideo    

  Home Team Name  Home Team Goals  Away Team Goals Away Team Name  \
0         France              4.0              1.0         Mexico   
1            USA              3.0              0.0        Belgium   
2     Yugoslavia              2.0              1.0         Brazil   
3        Romania              3.0              1.0           Peru   
4      Argentina              1.0              0.0         France   

  Win conditions  Attendance  Half-time Home Goals  Half-time Away Goals  \
0                     4444.0                   3.0                   0.0   
1                    18346.0                   2.0                   0.0   
2                    24059.0                   2.0                   0.0   
3                     2549.0                   1.0                   0.0   
4                    23409.0                   0.0                   0.0   

                  Referee               Assistant 1  \
0  LOMBARDI Domingo (URU)     CRISTOPHE Henry (BEL)   
1       MACIAS Jose (ARG)  MATEUCCI Francisco (URU)   
2     TEJADA Anibal (URU)   VALLARINO Ricardo (URU)   
3   WARNKEN Alberto (CHI)       LANGENUS Jean (BEL)   
4     REGO Gilberto (BRA)      SAUCEDO Ulises (BOL)   

                  Assistant 2  RoundID  MatchID Home Team Initials  \
0         REGO Gilberto (BRA)    201.0   1096.0                FRA   
1       WARNKEN Alberto (CHI)    201.0   1090.0                USA   
2         BALWAY Thomas (FRA)    201.0   1093.0                YUG   
3    MATEUCCI Francisco (URU)    201.0   1098.0                ROU   
4  RADULESCU Constantin (ROU)    201.0   1085.0                ARG   

  Away Team Initials  
0                MEX  
1                BEL  
2                BRA  
3                PER  
4                FRA

### calculting the unique values

In [21]:
df1['Stage'].unique()

array(['Group 1', 'Group 4', 'Group 2', 'Group 3', 'Semi-finals', 'Final',
       'Preliminary round', 'Quarter-finals', 'Match for third place',
       'First round', 'Group 6', 'Group B', 'Group A', 'Group 5',
       'Group C', 'Group D', 'Group F', 'Group E', 'Round of 16',
       'Group H', 'Group G', 'Third place', 'Play-off for third place'],
      dtype=object)

In [22]:
df1['Stadium'].unique()

array(['Pocitos', 'Parque Central', 'Estadio Centenario',
       'Stadio Benito Mussolini', 'Giorgio Ascarelli', 'San Siro',
       'Littorale', 'Giovanni Berta', 'Luigi Ferraris', 'Nazionale PNF',
       'Littorio', 'Parc des Princes', 'Velodrome Municipale',
       'Stade Olympique', 'Stade Municipal', 'Stade V�lodrome',
       'Stade de la Meinau', 'Cavee Verte', 'Stade du Parc Lescure',
       'Victor Boucquey', 'Fort Carree',
       'Maracan� - Est�dio Jornalista M�rio Filho', 'Durival de Brito',
       'Pacaembu', 'Independencia', 'Eucaliptos', 'Ilha do Retiro',
       'Wankdorf Stadium', 'Hardturm', 'Charmilles', 'La Pontaise',
       'St. Jakob', 'Comunale di Cornaredo', 'Rasunda Stadium',
       'Nya Ullevi', 'Malmo Stadion', 'Jarnvallen', 'Idrottsparken',
       'Arosvallen', 'Rimnersvallen', 'Orjans Vall', 'Ryavallen',
       'Olympia Stadium', 'Tunavallen', 'Eyravallen', 'Carlos Dittborn',
       'Estadio Sausalito', 'Estadio El Teniente-Codelco', 'Nacional',
       'Wemble

In [23]:
df1['Stadium'] = df1['Stadium']. replace(['Stade V�lodrome','Maracan� - Est�dio Jornalista M�rio Filho',
                                          'Nou Camp - Estadio Le�n','Estadio Jos� Mar�a Minella'],
                                   ['Stade Vlodrome','Estdio Jornalista Mrio Filho','Nou Camp - Estadio Len',
                                    'Estadio Jos Mara Minella'])

In [24]:
df1['Stadium'].unique()

array(['Pocitos', 'Parque Central', 'Estadio Centenario',
       'Stadio Benito Mussolini', 'Giorgio Ascarelli', 'San Siro',
       'Littorale', 'Giovanni Berta', 'Luigi Ferraris', 'Nazionale PNF',
       'Littorio', 'Parc des Princes', 'Velodrome Municipale',
       'Stade Olympique', 'Stade Municipal', 'Stade Vlodrome',
       'Stade de la Meinau', 'Cavee Verte', 'Stade du Parc Lescure',
       'Victor Boucquey', 'Fort Carree', 'Estdio Jornalista Mrio Filho',
       'Durival de Brito', 'Pacaembu', 'Independencia', 'Eucaliptos',
       'Ilha do Retiro', 'Wankdorf Stadium', 'Hardturm', 'Charmilles',
       'La Pontaise', 'St. Jakob', 'Comunale di Cornaredo',
       'Rasunda Stadium', 'Nya Ullevi', 'Malmo Stadion', 'Jarnvallen',
       'Idrottsparken', 'Arosvallen', 'Rimnersvallen', 'Orjans Vall',
       'Ryavallen', 'Olympia Stadium', 'Tunavallen', 'Eyravallen',
       'Carlos Dittborn', 'Estadio Sausalito',
       'Estadio El Teniente-Codelco', 'Nacional', 'Wembley Stadium',
       'H

In [25]:
df1['City'].unique()

array(['Montevideo ', 'Turin ', 'Naples ', 'Milan ', 'Bologna ',
       'Florence ', 'Genoa ', 'Rome ', 'Trieste ', 'Paris ', 'Reims ',
       'Colombes ', 'Toulouse ', 'Marseilles ', 'Strasbourg ',
       'Le Havre ', 'Bordeaux ', 'Lille ', 'Antibes ', 'Rio De Janeiro ',
       'Curitiba ', 'Sao Paulo ', 'Belo Horizonte ', 'Porto Alegre ',
       'Recife ', 'Berne ', 'Zurich ', 'Geneva ', 'Lausanne ', 'Basel ',
       'Lugano ', 'Solna ', 'Gothenburg ', 'Malm� ', 'Sandviken ',
       'Norrk�Ping ', 'Vasteras ', 'Udevalla ', 'Halmstad ', 'Boras ',
       'Helsingborg ', 'Eskilstuna ', 'Orebro ', 'Arica ',
       'Vina Del Mar ', 'Rancagua ', 'Santiago De Chile ', 'London ',
       'Sheffield ', 'Liverpool ', 'Middlesbrough ', 'Manchester ',
       'Birmingham ', 'Sunderland ', 'Mexico City ', 'Puebla ', 'Leon ',
       'Guadalajara ', 'Toluca ', 'Frankfurt/Main ', 'Berlin West ',
       'Hamburg ', 'Dortmund ', 'Hanover ', 'D�Sseldorf ', 'Munich ',
       'Stuttgart ', 'Gelsenkirchen '

In [26]:
df1['City'] = df1['City']. replace(['Norrk�Ping ','La Coru�A ','D�Sseldorf '],
                                   ['Norrk Ping','La Coru A', 'D Sseldrof'])

In [27]:
df1['City'].unique()

array(['Montevideo ', 'Turin ', 'Naples ', 'Milan ', 'Bologna ',
       'Florence ', 'Genoa ', 'Rome ', 'Trieste ', 'Paris ', 'Reims ',
       'Colombes ', 'Toulouse ', 'Marseilles ', 'Strasbourg ',
       'Le Havre ', 'Bordeaux ', 'Lille ', 'Antibes ', 'Rio De Janeiro ',
       'Curitiba ', 'Sao Paulo ', 'Belo Horizonte ', 'Porto Alegre ',
       'Recife ', 'Berne ', 'Zurich ', 'Geneva ', 'Lausanne ', 'Basel ',
       'Lugano ', 'Solna ', 'Gothenburg ', 'Malm� ', 'Sandviken ',
       'Norrk Ping', 'Vasteras ', 'Udevalla ', 'Halmstad ', 'Boras ',
       'Helsingborg ', 'Eskilstuna ', 'Orebro ', 'Arica ',
       'Vina Del Mar ', 'Rancagua ', 'Santiago De Chile ', 'London ',
       'Sheffield ', 'Liverpool ', 'Middlesbrough ', 'Manchester ',
       'Birmingham ', 'Sunderland ', 'Mexico City ', 'Puebla ', 'Leon ',
       'Guadalajara ', 'Toluca ', 'Frankfurt/Main ', 'Berlin West ',
       'Hamburg ', 'Dortmund ', 'Hanover ', 'D Sseldrof', 'Munich ',
       'Stuttgart ', 'Gelsenkirchen ', 

In [28]:
df1['Home Team Name'].unique()

array(['France', 'USA', 'Yugoslavia', 'Romania', 'Argentina', 'Chile',
       'Uruguay', 'Brazil', 'Paraguay', 'Austria', 'Hungary',
       'Switzerland', 'Sweden', 'Germany', 'Spain', 'Italy',
       'Czechoslovakia', 'Cuba', 'England', 'Germany FR', 'Turkey',
       'Soviet Union', 'Northern Ireland', 'Mexico', 'Wales', 'Portugal',
       'Korea DPR', 'Peru', 'Belgium', 'Bulgaria', 'German DR', 'Zaire',
       'Poland', 'Australia', 'Scotland', 'Netherlands', 'Haiti',
       'Tunisia', 'Algeria', 'Honduras', 'Canada', 'Morocco',
       'Korea Republic', 'Iraq', 'Denmark', 'rn">United Arab Emirates',
       'Costa Rica', 'Cameroon', 'rn">Republic of Ireland', 'Colombia',
       'Norway', 'Nigeria', 'Saudi Arabia', 'Bolivia', 'Russia', 'Greece',
       'Jamaica', 'South Africa', 'Japan', 'Croatia', 'China PR',
       'Senegal', 'Slovenia', 'Ecuador', 'rn">Trinidad and Tobago',
       'rn">Serbia and Montenegro', 'Angola', 'Czech Republic', 'Togo',
       'Iran', "C�te d'Ivoire", 'Ghana

In [29]:
df1['Home Team Name'] = df1['Home Team Name']. replace([ "C�te d'Ivoire"],
                                   ["C te d' Ivoire"])

In [30]:
df1['Home Team Name'].unique()

array(['France', 'USA', 'Yugoslavia', 'Romania', 'Argentina', 'Chile',
       'Uruguay', 'Brazil', 'Paraguay', 'Austria', 'Hungary',
       'Switzerland', 'Sweden', 'Germany', 'Spain', 'Italy',
       'Czechoslovakia', 'Cuba', 'England', 'Germany FR', 'Turkey',
       'Soviet Union', 'Northern Ireland', 'Mexico', 'Wales', 'Portugal',
       'Korea DPR', 'Peru', 'Belgium', 'Bulgaria', 'German DR', 'Zaire',
       'Poland', 'Australia', 'Scotland', 'Netherlands', 'Haiti',
       'Tunisia', 'Algeria', 'Honduras', 'Canada', 'Morocco',
       'Korea Republic', 'Iraq', 'Denmark', 'rn">United Arab Emirates',
       'Costa Rica', 'Cameroon', 'rn">Republic of Ireland', 'Colombia',
       'Norway', 'Nigeria', 'Saudi Arabia', 'Bolivia', 'Russia', 'Greece',
       'Jamaica', 'South Africa', 'Japan', 'Croatia', 'China PR',
       'Senegal', 'Slovenia', 'Ecuador', 'rn">Trinidad and Tobago',
       'rn">Serbia and Montenegro', 'Angola', 'Czech Republic', 'Togo',
       'Iran', "C te d' Ivoire", 'Ghan

In [31]:
df1['Away Team Name'].unique()

array(['Mexico', 'Belgium', 'Brazil', 'Peru', 'France', 'Bolivia',
       'Paraguay', 'Romania', 'Chile', 'USA', 'Yugoslavia', 'Argentina',
       'Egypt', 'Netherlands', 'Switzerland', 'Sweden', 'Spain',
       'Hungary', 'Austria', 'Germany', 'Czechoslovakia',
       'Dutch East Indies', 'Norway', 'Poland', 'Cuba', 'Italy',
       'England', 'Scotland', 'Turkey', 'Korea Republic', 'Germany FR',
       'Uruguay', 'Wales', 'Northern Ireland', 'Soviet Union', 'Colombia',
       'Bulgaria', 'Korea DPR', 'Portugal', 'Israel', 'Morocco',
       'El Salvador', 'Australia', 'Haiti', 'German DR', 'Zaire',
       'IR Iran', 'Tunisia', 'Iran', 'Cameroon', 'New Zealand', 'Algeria',
       'Honduras', 'Kuwait', 'Iraq', 'Denmark', 'Canada',
       'rn">Republic of Ireland', 'rn">United Arab Emirates',
       'Costa Rica', 'Saudi Arabia', 'Russia', 'Greece', 'Nigeria',
       'South Africa', 'Japan', 'Croatia', 'Jamaica', 'Senegal',
       'Slovenia', 'Ecuador', 'China PR', "C�te d'Ivoire",
       

In [32]:
df1['Away Team Name'] = df1['Away Team Name']. replace([ "C�te d'Ivoire"],
                                   ["C te d' Ivoire"])

In [33]:
df1['Away Team Name'].unique()

array(['Mexico', 'Belgium', 'Brazil', 'Peru', 'France', 'Bolivia',
       'Paraguay', 'Romania', 'Chile', 'USA', 'Yugoslavia', 'Argentina',
       'Egypt', 'Netherlands', 'Switzerland', 'Sweden', 'Spain',
       'Hungary', 'Austria', 'Germany', 'Czechoslovakia',
       'Dutch East Indies', 'Norway', 'Poland', 'Cuba', 'Italy',
       'England', 'Scotland', 'Turkey', 'Korea Republic', 'Germany FR',
       'Uruguay', 'Wales', 'Northern Ireland', 'Soviet Union', 'Colombia',
       'Bulgaria', 'Korea DPR', 'Portugal', 'Israel', 'Morocco',
       'El Salvador', 'Australia', 'Haiti', 'German DR', 'Zaire',
       'IR Iran', 'Tunisia', 'Iran', 'Cameroon', 'New Zealand', 'Algeria',
       'Honduras', 'Kuwait', 'Iraq', 'Denmark', 'Canada',
       'rn">Republic of Ireland', 'rn">United Arab Emirates',
       'Costa Rica', 'Saudi Arabia', 'Russia', 'Greece', 'Nigeria',
       'South Africa', 'Japan', 'Croatia', 'Jamaica', 'Senegal',
       'Slovenia', 'Ecuador', 'China PR', "C te d' Ivoire",
      

In [34]:
df1['Home Team Name'].unique()

array(['France', 'USA', 'Yugoslavia', 'Romania', 'Argentina', 'Chile',
       'Uruguay', 'Brazil', 'Paraguay', 'Austria', 'Hungary',
       'Switzerland', 'Sweden', 'Germany', 'Spain', 'Italy',
       'Czechoslovakia', 'Cuba', 'England', 'Germany FR', 'Turkey',
       'Soviet Union', 'Northern Ireland', 'Mexico', 'Wales', 'Portugal',
       'Korea DPR', 'Peru', 'Belgium', 'Bulgaria', 'German DR', 'Zaire',
       'Poland', 'Australia', 'Scotland', 'Netherlands', 'Haiti',
       'Tunisia', 'Algeria', 'Honduras', 'Canada', 'Morocco',
       'Korea Republic', 'Iraq', 'Denmark', 'rn">United Arab Emirates',
       'Costa Rica', 'Cameroon', 'rn">Republic of Ireland', 'Colombia',
       'Norway', 'Nigeria', 'Saudi Arabia', 'Bolivia', 'Russia', 'Greece',
       'Jamaica', 'South Africa', 'Japan', 'Croatia', 'China PR',
       'Senegal', 'Slovenia', 'Ecuador', 'rn">Trinidad and Tobago',
       'rn">Serbia and Montenegro', 'Angola', 'Czech Republic', 'Togo',
       'Iran', "C te d' Ivoire", 'Ghan

In [35]:
df1['Home Team Name'] = df1['Home Team Name']. replace(['''rn">Trinidad and Tobago''',"C te d' Ivoire",
                                                        'rn">Bosnia and Herzegovina'],
                                   ['Trinidad and Tobago','Ctedivore','Bosnia and Herzegovina'])

In [36]:
df1['Home Team Name'].unique()

array(['France', 'USA', 'Yugoslavia', 'Romania', 'Argentina', 'Chile',
       'Uruguay', 'Brazil', 'Paraguay', 'Austria', 'Hungary',
       'Switzerland', 'Sweden', 'Germany', 'Spain', 'Italy',
       'Czechoslovakia', 'Cuba', 'England', 'Germany FR', 'Turkey',
       'Soviet Union', 'Northern Ireland', 'Mexico', 'Wales', 'Portugal',
       'Korea DPR', 'Peru', 'Belgium', 'Bulgaria', 'German DR', 'Zaire',
       'Poland', 'Australia', 'Scotland', 'Netherlands', 'Haiti',
       'Tunisia', 'Algeria', 'Honduras', 'Canada', 'Morocco',
       'Korea Republic', 'Iraq', 'Denmark', 'rn">United Arab Emirates',
       'Costa Rica', 'Cameroon', 'rn">Republic of Ireland', 'Colombia',
       'Norway', 'Nigeria', 'Saudi Arabia', 'Bolivia', 'Russia', 'Greece',
       'Jamaica', 'South Africa', 'Japan', 'Croatia', 'China PR',
       'Senegal', 'Slovenia', 'Ecuador', 'Trinidad and Tobago',
       'rn">Serbia and Montenegro', 'Angola', 'Czech Republic', 'Togo',
       'Iran', 'Ctedivore', 'Ghana', 'Ukra

### EDA on dataset 1

In [37]:
import matplotlib.pyplot as plt
import seaborn as sns
import plotly as py                
import cufflinks as cf
%matplotlib inline
from plotly.offline import iplot   
py.offline.init_notebook_mode(connected=True)
cf.go_offline()


In [40]:
df1['Year'] = pd.to_datetime(df1['Year'], format='%Y') 

In [50]:
year_with_stages = df1[['Year','Stage']].value_counts()
year_with_stages

Year        Stage                
2014-01-01  Round of 16              14
1938-01-01  First round               9
1982-01-01  Group 4                   9
            Group 3                   9
            Group 2                   9
                                     ..
1966-01-01  Match for third place     1
            Final                     1
1962-01-01  Match for third place     1
            Final                     1
1930-01-01  Final                     1
Length: 184, dtype: int64

In [47]:
year_with_stages.iplot(kind='bar', title='Stages of match during the year')

From the above we can see that in the year 2014 height number of matches was played and it belong to Round of 16 

In [52]:
import plotly_express as px

In [55]:
# In which stage the height matches was played.
df1[['Stage']].value_counts().iplot(kind='bar')

4572


In [59]:
# In which city the most stadium persent
city_with_most_stadium = df1[['Stadium','City']].value_counts()
city_with_most_stadium.iplot(kind='bar', title='Most stadium in city')

From the above we can see that Estadio Azteca, Mexico City has most stadium

In [66]:
df1[['Home Team Name','Home Team Goals']].value_counts().iplot(kind='bar', title='Most Goal by home team')

From the above we can see that Italy played most of the matches in there home ground.

In [69]:
# Which home team hit most of the goal while playing with other teams
df1[['Home Team Name','Away Team Goals']].value_counts()

Home Team Name            Away Team Goals
Brazil                    0.0                35
                          1.0                32
Italy                     1.0                28
England                   0.0                23
Argentina                 0.0                23
                                             ..
Germany                   3.0                 1
Germany FR                3.0                 1
Ghana                     2.0                 1
Peru                      2.0                 1
rn">United Arab Emirates  2.0                 1
Length: 214, dtype: int64

In [71]:
df1[['Home Team Name','Away Team Goals']].value_counts().iplot(kind='scatter', title='Most goal with other team')

Brazil hit most of the goal while playing with other team in there home ground.

In [73]:
# Which team has hit most of the goal while playing with other team and visevers
df1[['Home Team Goals','Away Team Goals','Away Team Name']].value_counts()

Home Team Goals  Away Team Goals  Away Team Name
0.0              0.0              Argentina         6
2.0              1.0              Mexico            6
                                  Spain             6
                                  Italy             5
1.0              0.0              England           5
                                                   ..
                 2.0              Germany FR        1
                                  Ecuador           1
                                  Costa Rica        1
                                  Belgium           1
10.0             1.0              El Salvador       1
Length: 558, dtype: int64

In [74]:
df1[['Home Team Goals','Away Team Goals','Away Team Name']].value_counts().iplot(kind='bar')

Argentina has the height count

In [81]:
# What is the most winning condition 
df1['Win conditions'].value_counts().iplot(kind='bar', title='Winning condition')

Italy win most of the time after extra time

In [84]:
df1.head()
# in which stadium the attendence was very high
df1[['Stadium','Attendance']].value_counts().iplot(kind='line', title='Most attendence')

We can see that Parc des Princees was the stadium and the count was 45500

In [95]:

# What is the count of half team goal 
df1[['Home Team Name','Half-time Home Goals']].value_counts().iplot(kind='bar',title='Goal in half time')

We see that Brazil hit most goal in half time

In [96]:
df1[['Away Team Name','Half-time Away Goals']].value_counts().iplot(kind='bar',title='Goal in half time')

Mexico hit most goal in half time

## Summary 

From all the above plots we can observe that its a clear comparision between the stadiumns, groups, teams half time.

In [97]:
df2.head()

RoundID  MatchID Team Initials           Coach Name Line-up  Shirt Number  \
0      201     1096           FRA  CAUDRON Raoul (FRA)       S             0   
1      201     1096           MEX     LUQUE Juan (MEX)       S             0   
2      201     1096           FRA  CAUDRON Raoul (FRA)       S             0   
3      201     1096           MEX     LUQUE Juan (MEX)       S             0   
4      201     1096           FRA  CAUDRON Raoul (FRA)       S             0   

        Player Name Position Event  
0       Alex THEPOT       GK   NaN  
1   Oscar BONFIGLIO       GK   NaN  
2  Marcel LANGILLER      NaN  G40'  
3      Juan CARRENO      NaN  G70'  
4   Ernest LIBERATI      NaN   NaN

In [98]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 37784 entries, 0 to 37783
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   RoundID        37784 non-null  int64 
 1   MatchID        37784 non-null  int64 
 2   Team Initials  37784 non-null  object
 3   Coach Name     37784 non-null  object
 4   Line-up        37784 non-null  object
 5   Shirt Number   37784 non-null  int64 
 6   Player Name    37784 non-null  object
 7   Position       4143 non-null   object
 8   Event          9069 non-null   object
dtypes: int64(3), object(6)
memory usage: 2.6+ MB


In [99]:
df2.isnull().sum()

RoundID              0
MatchID              0
Team Initials        0
Coach Name           0
Line-up              0
Shirt Number         0
Player Name          0
Position         33641
Event            28715
dtype: int64

In [100]:
# Percentage of missing value
percentage = df2.isnull().sum()/len(df2) * 100

In [101]:
percentage

RoundID           0.000000
MatchID           0.000000
Team Initials     0.000000
Coach Name        0.000000
Line-up           0.000000
Shirt Number      0.000000
Player Name       0.000000
Position         89.035041
Event            75.997777
dtype: float64

In [102]:
# Let's drop the missing values
df2.dropna(inplace=True)

In [103]:
df2.drop_duplicates(inplace=True)

In [104]:
df2.shape

(555, 9)

In [105]:
df2.head()

RoundID  MatchID Team Initials                 Coach Name Line-up  \
35       201     1090           USA           MILLAR Bob (USA)       S   
74       201     1093           BRA  DE CARVALHO Pindaro (BRA)       S   
113      201     1098           PER        BRU Francisco (ESP)       S   
415      201     1091           BRA  DE CARVALHO Pindaro (BRA)       S   
468      201     1089           PAR   DURAND LAGUNA Jose (ARG)       S   

     Shirt Number       Player Name Position      Event  
35              0        Tom FLORIE        C       G45'  
74              0         PREGUINHO        C       G62'  
113             0   Placido GALINDO        C       R70'  
415             0         PREGUINHO        C  G67' G83'  
468             0  Luis VARGAS PENA        C       G40'

In [106]:
df2.reset_index(inplace=True)

In [107]:
df2.head()

index  RoundID  MatchID Team Initials                 Coach Name Line-up  \
0     35      201     1090           USA           MILLAR Bob (USA)       S   
1     74      201     1093           BRA  DE CARVALHO Pindaro (BRA)       S   
2    113      201     1098           PER        BRU Francisco (ESP)       S   
3    415      201     1091           BRA  DE CARVALHO Pindaro (BRA)       S   
4    468      201     1089           PAR   DURAND LAGUNA Jose (ARG)       S   

   Shirt Number       Player Name Position      Event  
0             0        Tom FLORIE        C       G45'  
1             0         PREGUINHO        C       G62'  
2             0   Placido GALINDO        C       R70'  
3             0         PREGUINHO        C  G67' G83'  
4             0  Luis VARGAS PENA        C       G40'

In [115]:
# Let's check the team player name with shirt number 
df2[['Player Name','Shirt Number']].value_counts().iplot(kind='bar')

From the above we can see that number on there tshirt

In [122]:
df2['Team Initials'].value_counts().iplot(kind = 'bar')

Most of the time team Initials have change for ARG

## Summary: 

We can only see the height team initial change , tshirt number 

## Data Frme 3

In [123]:
df3.head()

Year      Country      Winner      Runners-Up    Third      Fourth  \
0  1930      Uruguay     Uruguay       Argentina      USA  Yugoslavia   
1  1934        Italy       Italy  Czechoslovakia  Germany     Austria   
2  1938       France       Italy         Hungary   Brazil      Sweden   
3  1950       Brazil     Uruguay          Brazil   Sweden       Spain   
4  1954  Switzerland  Germany FR         Hungary  Austria     Uruguay   

   GoalsScored  QualifiedTeams  MatchesPlayed Attendance  
0           70              13             18    590.549  
1           70              16             17    363.000  
2           84              15             18    375.700  
3           88              13             22  1.045.246  
4          140              16             26    768.607

In [130]:
df3[['Year','Country','Winner']]

Year       Country      Winner
0   1930       Uruguay     Uruguay
1   1934         Italy       Italy
2   1938        France       Italy
3   1950        Brazil     Uruguay
4   1954   Switzerland  Germany FR
5   1958        Sweden      Brazil
6   1962         Chile      Brazil
7   1966       England     England
8   1970        Mexico      Brazil
9   1974       Germany  Germany FR
10  1978     Argentina   Argentina
11  1982         Spain       Italy
12  1986        Mexico   Argentina
13  1990         Italy  Germany FR
14  1994           USA      Brazil
15  1998        France      France
16  2002   Korea/Japan      Brazil
17  2006       Germany       Italy
18  2010  South Africa       Spain
19  2014        Brazil     Germany

In [136]:
# Based on the year which team was the winner
fig = px.bar(df3, x='Winner',color='Country')
fig.show()

From the above we can see that Brazil won most of the match with different countries such as Sweden,Chile,Mexico, and soon

In [142]:
# Based on the year which team was the winner
df3[['Runners-Up']].iplot(kind='bar')

From the above we can see that France was runner for the most sof the time

In [144]:
df3.head()

Year      Country      Winner      Runners-Up    Third      Fourth  \
0  1930      Uruguay     Uruguay       Argentina      USA  Yugoslavia   
1  1934        Italy       Italy  Czechoslovakia  Germany     Austria   
2  1938       France       Italy         Hungary   Brazil      Sweden   
3  1950       Brazil     Uruguay          Brazil   Sweden       Spain   
4  1954  Switzerland  Germany FR         Hungary  Austria     Uruguay   

   GoalsScored  QualifiedTeams  MatchesPlayed Attendance  
0           70              13             18    590.549  
1           70              16             17    363.000  
2           84              15             18    375.700  
3           88              13             22  1.045.246  
4          140              16             26    768.607

In [150]:
# in which year height goal was scored by which country
px.scatter(df3, x='Year',y='GoalsScored', color='Country')

France Scored the height gaol in the year 2000

In [151]:
# in which year team qualified the most as pe color
px.scatter(df3, x='Year',y='QualifiedTeams', color='Country')

from the above we can see that France in the year 1998, Japan in 2002 , Germany, South africa, Brazil

In [152]:
df3.head()

Year      Country      Winner      Runners-Up    Third      Fourth  \
0  1930      Uruguay     Uruguay       Argentina      USA  Yugoslavia   
1  1934        Italy       Italy  Czechoslovakia  Germany     Austria   
2  1938       France       Italy         Hungary   Brazil      Sweden   
3  1950       Brazil     Uruguay          Brazil   Sweden       Spain   
4  1954  Switzerland  Germany FR         Hungary  Austria     Uruguay   

   GoalsScored  QualifiedTeams  MatchesPlayed Attendance  
0           70              13             18    590.549  
1           70              16             17    363.000  
2           84              15             18    375.700  
3           88              13             22  1.045.246  
4          140              16             26    768.607

In [155]:
df3[['Winner','Runners-Up','Third','Fourth']].iplot(kind='bar')

From the above we can see the comparision of 'Winner','Runners-Up','Third','Fourth' in every match

In [162]:
px.histogram(df3, x='Year', y='Attendance')

We can see that in the year 2000 the count of attendence was very high

In [164]:
px.scatter(df3, x='MatchesPlayed', y = 'Winner',color='Country')

Most of the matches was played between Spain and South Africa

## Summary 

From all the above we can see that who won most of the time, count of people was huge in which year, who was runner, winner, third and fourth position etc.